controllers/cluster_controller.go

```diff
@@ -1188,35 +1222,42 @@ func chooseDistributedProcesses(processes []localityInfo, count int, constraint
 		currentLimits[field] = 1
 	}
 
+	// Sort the processes to ensure a deterministic result
+	sortLocalities(processes)
+
 	for len(chosen) < count {
 		choseAny := false
 
 		for _, process := range processes {
-			if !chosenIDs[process.ID] {
```

Comment:
Is anything changing in the behavior of this section, or is this just rearranging the code to reduce nesting?

Reply:
I only reduced the nesting to make it more readable.